### PR TITLE
!!!TASK: Migrate extracted NodeTypes from Neos.NodeTypes to subpackage namespace

### PR DESCRIPTION
**Pull request by mficzel**
Currently Neos.NodeTypes defines many nodes of subpackages abstract and extends them in its own namespace. This makes it hard to support setups with Neos.NodeTypes or only the subpackages from a single site-package. 

This is especially problematic since we are not recommending Neos.NodeTypes anymore but still have it in the development collection.

This changes adds a migration to convert all NodeTypes that were extracted from Neos.NodeTypes into the namespace of the respective subpackage.

To apply the migration run `./flow node:migrate 20190304111200` 

Note: With this change we can remove the dependency to Neos.NodeTypes from Neos.Demo and only require the subpackages we really need.

### CODE REVIEW

**Comment by mficzel** [comment]
It has to be decided wether we consider this breaky and has to go int 5.0 instead of 4.3 since it contains a node-migration. However existing nodes should still render but cannot be created any more in Neos.NodeTypes namespace.

**Comment by rolandschuetz** [comment]
This is a breaking change and should go into Neos 5. 
When refactoring the NodeTypes package into separate subpackages it was necessary to do this abstract:true and basically inheriting the same nodetype under a different namespace to not break existing code.

Most people who use the NodeTypes package directly extend and override the NodeTypes (as recommended a few years ago). So migrating existing NodeTypes will most likely break exisiting websites.

**Comment by mficzel** [comment]
@rolandschuetz agreed i marked this for next major. Do you think we should get rid of the overrides in that case directly or schould we still keep them?

**Comment by rolandschuetz** [comment]
I would love to get rid of them. Using the general NodeTypes package should
be be discouraged. Maybe we remove it all together?

All the valuable parts are already abstracted in seoarate packages, the the
only poinz right now for this package is to not break existing code. If we
are going to do this in the next major, let get rid of the whole package.

Martin Ficzel <notifications@github.com> schrieb am Mo., 4. März 2019,
12:58:

**Comment by mficzel** [comment]
@rolandschuetz i that case we would not have a place where to put the migration.

**Comment by rolandschuetz** [comment]
Hmm... so we could only deprecate in Neos 5 and remove in 6?

Martin Ficzel <notifications@github.com> schrieb am Mo., 4. März 2019,
14:50:

**Comment by mficzel** [comment]
@rolandschuetz added the deprecation notice you requested

**Comment by daniellienert** [comment]
@mficzel: Sadly we have a merge conflict now. Would you please solve it?

File: Neos.NodeTypes/Migrations/TYPO3CR/Version20190304111200.yaml [hunk 1]
```diff
@@ -0,0 +1,217 @@
+up:
```
Comment (daniellienert) [inline Neos.NodeTypes/Migrations/TYPO3CR/Version20190304111200.yaml]: New Migrations should be added to `ContentRepository`- TYPO3CR is deprecated

**Comment by mficzel** [comment]
Was rebased a while ago already and is mergable again.

File: Neos.NodeTypes/Migrations/TYPO3CR/Version20190304111200.yaml (hunk 1 above)
Comment (mficzel) [reply Neos.NodeTypes/Migrations/TYPO3CR/Version20190304111200.yaml]: @daniellienert this was adjusted.

**Comment by mficzel** [comment]
@daniellienert was rebased and adjusted. Please re review.

**Comment by mficzel** [comment]
Just a note: If this gets merged for 5.0 the Neos Demo needs to run the included migrations and default xml. I can take care once this is in.

**Comment by kdambekalns** [comment]
@mficzel Just talked to @daniellienert and he'll update the Neos.Demo package as needed.

**Pull request merged**